crates/polars-plan/src/dsl/function_expr/temporal.rs

```diff
@@ -95,91 +94,61 @@ pub(super) fn datetime(
     }
     let year = year.i32()?;
 
-    let mut month = month.cast(&DataType::UInt32)?;
+    let mut month = month.cast(&DataType::Int8)?;
```

Comment:
why does this need to change?

Reply:
Hi Marco--it doesn't, but I was trying to be consistent elsewhere where for e.g. `ca.month()` returns a `i8`, so now that we have `new_from_parts(...)` it made sense to me that the `month` input should be a `i8` as well.

I understand that it pretty much immediately gets cast back to an `i32` so I'll revert this minor change.

Reply:
@MarcoGorelli what's your opinion here? Should I revert and keep it to u32?